src/query/api/v1/handler/prometheus/common.go

```diff
@@ -47,15 +47,20 @@ const (
 	filterNameTagsParam = "tag"
 	errFormatStr        = "error parsing param: %s, error: %v"
 
-	// TODO: get timeouts from configs
-	maxTimeout     = time.Minute
-	defaultTimeout = time.Second * 15
+	maxTimeout = 5 * time.Minute
+	// DefaultTimeout is the default timeout for fetch requests
+	DefaultTimeout = 15 * time.Second
```

Comment:
It feels a bit weird to export this from here; I think it would read better if you had this hardcoded backup value in `src/query/api/v1/httpd/handler.go` and keep it unexported.

Then in `common_test` hardcode the 15 seconds as the test value (might be better to up to 30 in light of user comments on gitter?)

Reply:
Yeah I went back and forth on this. I'll put in `handler.go`